### DISCUSSION TITLE
[Feature] Add advanced mode in context search to show the vector distance

### DISCUSSION THREAD

**Discussion by VoVAllen** [immich-app/immich#8810] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I see many users feeling confused about the search results, thinking that the results include many irrelevant photos. This is because the current search is more like relevance sorting rather than searching. 

The challenge here is that we do not know how to truncate the results to ensure that only relevant results appear. I am thinking of adding a feature on the search interface that allows users to see or export the vector distance data and report it to the Immich community (purely voluntary) and with the ideal distance cutoff in their scenarios. It's more like a developer mode for users. 

Afterwards, we can review the data submitted by users to find a cutoff strategy. This will make that irrelevant results are excluded from the search (or users can adjust the cutoff distance to get more results).

Just some random thoughts on improving search function. Discussion welcomed!

Similar issue: https://github.com/immich-app/immich/issues/5996

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
I like it! In the future, UI for this could also be useful for #1673 - I can imagine a rule that applies based on a search query with a user-defined cutoff set.
